GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1551 @@
+                    DataServiceStreamResponse response =
+                        await photo.DownloadAsync();
+                    Console.WriteLine("\nUser {0} GOT thumbnailphoto", signedInUser.DisplayName);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError getting the user's photo - may not exist {0}",
+                        Program.ExtractErrorMessage(e));
+                }
+            }
+            return signedInUser;
+
+            #endregion
+        }
+
+        private static async Task UpdateSignedInUsersPhoto(IUser signedInUser)
+        {
+            #region update signed in user's photo
+
+            // NOTE:  Updating the signed in user's photo requires User.ReadWrite (when available) or 
+            // Directory.ReadWrite.All or Directory.AccessAsUser.All
+            if (signedInUser.ObjectId != null)
+            {
+                Console.WriteLine("\nDo you want to update your thumbnail photo? yes/no\n");
+                string update = Console.ReadLine();
+
+                if (update != null && update.Equals("yes"))
+                {
+                    try
+                    {
+                        Console.WriteLine("\nSpecify path of photo:");
+                        string photo = Console.ReadLine();
+                        //TODO - update with allowed art and save locally with project
+                        FileStream fileStream = new FileStream(photo.Trim('"'), FileMode.Open,
+                            FileAccess.Read);
+
+                        await signedInUser.ThumbnailPhoto.UploadAsync(fileStream, "application/image");
+                        Console.WriteLine("\nUser {0} was updated with a thumbnailphoto", signedInUser.DisplayName);
+                    }
+                    catch (Exception e)
+                    {
+                        Program.WriteError("\nError Updating the user photo {0}", Program.ExtractErrorMessage(e));
+                    }
+                }
+            }
+
+            #endregion
+
+            #endregion
+        }
+
+        private static async Task PeoplePickerExample(IActiveDirectoryClient client, string searchString)
+        {
+            #region People Picker example
+
+            //*********************************************************************
+            // People picker
+            // Search for a user using text string "Us" match against userPrincipalName, displayName, giveName, surname
+            // Requires minimum of User.ReadBasic.All.
+            //*********************************************************************
+
+            List<IUser> usersList = null;
+            IPagedCollection<IUser> searchResults = null;
+            try
+            {
+                IUserCollection userCollection = client.Users;
+                searchResults = await userCollection.Where(user =>
+                    user.UserPrincipalName.StartsWith(searchString) ||

```

Comment:
I'm actually getting an error/bug  here that says CurrentCultureIgnoreCase is not supported. The search appears to already be case insensitive.

---
In reply to: [81240382](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r81240382) [](ancestors = 81240382)